spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderMetrics.java

```diff
@@ -114,45 +124,38 @@ public void bindTo(MeterRegistry registry) {
 			String topic = topicInfo.getKey();
 			String group = topicInfo.getValue().getConsumerGroup();
 
+			//Schedule a task to compute the unconsumed messages for this group/topic every minute.
+			this.scheduler.scheduleWithFixedDelay(computeUnconsumedMessagesRunnable(topic, group, this.metadataConsumers),
+					10, DELAY_BETWEEN_TASK_EXECUTION, TimeUnit.SECONDS);
+
 			Gauge.builder(METRIC_NAME, this,
-					(o) -> computeUnconsumedMessages(topic, group)).tag("group", group)
+					(o) -> computeAndGetUnconsumedMessages(topic, group)).tag("group", group)
 					.tag("topic", topic)
 					.description("Unconsumed messages for a particular group and topic")
 					.register(registry);
 		}
 	}
 
-	private long computeUnconsumedMessages(String topic, String group) {
+	private Runnable computeUnconsumedMessagesRunnable(String topic, String group, Map<String, Consumer<?, ?>> metadataConsumers) {
+		return () -> {
+			long lag = 0;
+			try {
+				lag = findAnyLag(topic, group, lag, metadataConsumers);
+				this.unconsumedMessages.put(topic + "-" + group, lag);
+			}
+			catch (Exception ex) {
+				LOG.debug("Cannot generate metric for topic: " + topic, ex);
+			}
+		};
+	}
+
+	private long computeAndGetUnconsumedMessages(String topic, String group) {
 		ExecutorService exec = Executors.newSingleThreadExecutor();
```

Comment:
Creating and destroying an executor for each topic/group seems a little excessive. Can't we use a thread pool executor?